### DISCUSSION TITLE
Please help me to order nodes in counter clockwise direction

### DISCUSSION THREAD

**Discussion by kosirm** [hizzgdev/jsmind#565] [General]
Dear Mr. Zhigang Zhang, thank you so much for this library! It is very good.
I have one problem. I would like to order first level children (first level after root, e.g. nodes with direction(left or right) property) to be oriented counter clockwise on map import/export. By default map is exported like this: root node, then left nodes from top to bottom, then right nodes from top to bottom. What I would like to achieve is this: root node, then left nodes from top to bottom, then right nodes from bottom to top. I'm attaching images to explain my desired output in json file.
![default](https://github.com/hizzgdev/jsmind/assets/54502/a3cd1c08-1925-41b5-a423-7e29a41fec67)
![desired](https://github.com/hizzgdev/jsmind/assets/54502/09418cff-45df-4fe7-8a66-b49221fa83bf)

**Comment by hizzgdev** [comment]
Hi @kosirm , what do you mean `import` and `export`? 

With jsMind, the mindmap will be exported as is and can be imported back to its original layout, if you are referring to `import` and `export` as `jm.show(mindmap)` and `jm.get_data(data_format)`.

If you want to automatically layout nodes counterclockwise when you add them, I think this might not be possible because you have to artificially decide which node to start with and turn to the right.

If you can add nodes manually, you can try the following scripts, which allows you to position the nodes counterclockwise.


```javascript
jm.add_node('root', '1', '1', {}, 'left');
jm.add_node('root', '2', '2', {}, 'left');
jm.add_node('root', '3', '3', {}, 'left');
jm.add_node('root', '4', '4', {}, 'right');
jm.insert_node_before('4', '5', '5', {}, 'right');
jm.insert_node_before('5', '6', '6', {}, 'right');
```

<img width="230" alt="image" src="https://github.com/hizzgdev/jsmind/assets/1690290/f8077904-0097-4e6f-a473-48da865a2def">

**Reply by kosirm** [reply]
Thank you for your time! I can drag/drop nodes around, these numbers are just to illustrate what I want... Because if you save this file to jm format, which is json file, you will get this:
`{
    "meta": {
        "name": "jsMind",
        "author": "hizzgdev@163.com",
        "version": "0.8.0"
    },
    "format": "node_tree",
    "data": {
        "id": "root",
        "topic": "root",
        "expanded": true,
        "children": [
            {
                "id": "d0e66f16483dd868",
                "topic": "1",
                "expanded": true,
                "direction": "left"
            },
            {
                "id": "d0e66fc981710ec9",
                "topic": "2",
                "expanded": true,
                "direction": "left"
            },
            {
                "id": "d0e672788116c7f0",
                "topic": "3",
                "expanded": true,
                "direction": "left"
            },
            {
                "id": "d0e673fea3fb044b",
                "topic": "6",
                "expanded": true,
                "direction": "right"
            },
            {
                "id": "d0e67393e4354e16",
                "topic": "5",
                "expanded": true,
                "direction": "right"
            },
            {
                "id": "d0e6734cf61d5385",
                "topic": "4",
                "expanded": true,
                "direction": "right"
            }
        ]
    }
}`
So file is saved (and loaded) in this way:
<img width="230" alt="296822007-f8077904-0097-4e6f-a473-48da865a2def" src="https://github.com/hizzgdev/jsmind/assets/54502/83c8acd9-61da-4ace-b1a3-86cb905d2f2c">

There are many ways to read/write mind maps. Tony Buzan, creator of mindmapping is recommending clockwise, but my preferred way to make mind maps is counter-clockwise. I made hundreds of mindmaps during my life with pen and paper and I always make it counter-clockwise, my brain is simply like that 😄 

![Directions](https://github.com/hizzgdev/jsmind/assets/54502/27690e55-a20e-46d6-bb36-fac164d8bfd0)


But never mind, I decided, that I will make my mind map with vis.js. Anyway, congrats for your great work!

**Comment by hizzgdev** [comment]
Thanks @kosirm , I think I understand you now. What you expect is that the nodes are sorted in counterclockwise order in the json file.

We didn't pay much attention to the JSON content. If you'd like, could you please share why you care about it? do you need to import/export it from/to other mindmap app?

**Reply by kosirm** [reply]
Thanks for reply @hizzgdev , yes, exactly that. If I would be in charge I would make additional property for map called layout and then you can gradually add different layouts. But I don't know if anybody else is looking for that change, maybe it's only me... For example, what you sent me in image, if I save that to jm (json) and later load that json file in jsmind, it will be ordered by your default, which is LefDown-RightDown. However, I definitely decided that I will use vis.js network library, which is very much what I want, so good luck with your development! 😄
